### DISCUSSION TITLE
Command to quit steam big picture

### DISCUSSION THREAD

**Discussion by gt2416** [LizardByte/community#206] [Sunshine] [answered]
I had this working before, but after a recent update I lost all my application configs..
Basically when you end the session it quits steam big picture mode. It does not kill steam, just exits big picture mode. I forgot how I did it and now for the life of me I cant figure it out.

**Comment by Ebredvick** [comment, marked as answer]
I spent awhile looking for a solution to this same problem. I'm running windows 10 for reference.
I use nircmd, a tool you can download and add to your environment variables.
I then made two bat files. steam_open.bat and steam_close.bat and added them to my apps.json file for sunshine.

This solution basically tells sunshine to run the bat files and not Steam itself, so it has nothing to shutdown or quit out of, rather you start and shutdown steam Big Picture yourself with the detached command and undo command.

**steam_open.bat**
```
@echo off
start steam://open/bigpicture
```

**steam_close.bat**
```
@echo off
nircmd win close title "Steam Big Picture Mode"
```

**apps.json**
```
{
    "detached": [
        "path/to/your/file/steam_open.bat"
    ],
    "image-path": "steam.png",
    "name": "Steam Big Picture",
    "prep-cmd": [
        {
            "do": "",
            "undo": "path/to/your/file/steam_close.bat"
        }
    ]
}
```

**Comment by ld892012** [comment]
And make `-shutdown` flag to steam.exe not work ?
[Valve's Command line options](https://developer.valvesoftware.com/wiki/Command_line_options#Command-Line_Parameters_4)

**Reply by Ebredvick** [reply]
He prefers to keep Steam running even after ending his streaming session, ensuring it's always ready for use, similar to a gaming console. This approach prevents issues like delays when exiting Steam Big Picture Mode, which can take up to 30 seconds. By manually controlling Steam and using Sunshine to run only a batch script, the system feels more responsive.

**Comment by Loyola-London** [comment]
I had a similar requirement and this is a script I cobbled together. Doesn't require additional applications. Just list the file as a command in sunshine. The way this works, it launches big picture whether steam is running or not. Then, when you exit steam, the stream ends and steam restarts. 

```
@echo off
TASKLIST | FINDSTR "steam" && SET steam_running=true|| SET steam_running=false
IF "%steam_running%"=="true" (
    echo Steam is running
    goto startbigpicture
) ELSE (
    echo Steam is not running
    goto startsteam
)

:waitforshutdown
    TASKLIST | FINDSTR "steam" && SET still_running=true|| SET still_running=false
    IF "%still_running%"=="false" goto restartsteam
    timeout /t 1 /NOBREAK
    goto waitforshutdown

:startbigpicture
    start steam://open/bigpicture
    goto waitforshutdown
    goto end

:startsteam
    start /wait c:\"Program Files (x86)"\Steam\steam.exe -fulldesktopres -gamepadui
    goto waitforshutdown
    goto end

:restartsteam
    start c:\Program" "Files" "(x86)\Steam\steam.exe -fulldesktopres -start
    goto end

:end
    exit

```

**Reply by Ebredvick** [reply]
I did something like this before and called mine steam_watchdog. Ran into some issues with Steam hanging up sometimes, might've been something on my end though. Anyway, cool stuff you've got here, I bet it'll come in handy for someone since my solution involves downloading the nircmd tool.

**Comment by gongfuture** [comment]
I found steam big picture can be exited by entering "Alt+Enter", but I don't know how to implement it. I had tried use Sendkeys in vbs, but it didn't work. I can't get the real catch input window.

**Reply by droric** [reply]
Its actually as simple as running 
steam://close/bigpicture
to close big picture similarly to how running 
steam://open/bigpicture 
will open bigpicture.

**Comment by droric** [comment]
It's actually as simple as running 
steam://close/bigpicture
to close big picture similarly to how running 
steam://open/bigpicture 
will open bigpicture.  

Solution:  Just go into 'Applications' in Sunshine, edit Steam Big Picture, and add a Command Preparation.  Put "steam://close/bigpicture" (without quotes) in the Undo Command box.  Click save and apply, done.
